### PR TITLE
[Doc] Add advance doc for dataset

### PR DESCRIPTION
**Pull request by LeoXing1996**

### CODE REVIEW

File: docs/en/advanced_guides/datasets.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    img_b_path = self.data_infos_b[idx % self.len_b]['path']
+    results = dict()
+    results[f'img_{self.domain_a}_path'] = img_a_path
+    results[f'img_{self.domain_b}_path'] = img_b_path
+    return self.pipeline(results)
+```
+
+## 6. Add new dataset
+
+You may refer to [tutorial in MMEngine](https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/basedataset.md).
```
Comment (Z-Fran) [inline docs/en/advanced_guides/datasets.md:147]: This link is unavailable.

File: docs/en/advanced_guides/datasets.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Dataset
+
+In this document, we will introduce the design of each datasets in MMGeneration and how users to add their own dataset.
```
Comment (zengyh1900) [inline docs/en/advanced_guides/datasets.md:3]: .... and how can users design their own dataset.

File: docs/en/advanced_guides/datasets.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+In 1.x version of MMGeneration, all datasets are inherited from BaseDataset.
+Each dataset load the data list by `load_data_list` and call `prepare_data` in `__getitem__` to load the data info. In `prepare_data`, `get_data_info` and transforms pipeline will be called progressively.
```
Comment (zengyh1900) [inline docs/en/advanced_guides/datasets.md:6]: Specifically, a data processing pipeline in a dataset consists of the following steps: 
1. loading the list of dataset samples, implemented in `prepare_data`
2. loading dataset information, implemented in `__getitem__`
3. xxxxx

File: docs/en/advanced_guides/datasets.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+If you want to implement a new dataset class, you must implement `load_data_list` function. We also encourage users to use the original data loading logic provided by `BaseDataset`.
```
Comment (zengyh1900) [inline docs/en/advanced_guides/datasets.md:8]: you must implement `load_data_list` function
-->
you only need to implement `load_data_list` function

File: docs/en/advanced_guides/datasets.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+def prepare_test_data(self, idx):
+    img_a_path = self.data_infos_a[idx % self.len_a]['path']
+    img_b_path = self.data_infos_b[idx % self.len_b]['path']
+    results = dict()
+    results[f'img_{self.domain_a}_path'] = img_a_path
+    results[f'img_{self.domain_b}_path'] = img_b_path
+    return self.pipeline(results)
+```
+
+## 6. Add new dataset
```
Comment (zengyh1900) [inline docs/en/advanced_guides/datasets.md:145]: ## 6. Design your own dataset

**Pull request merged**